protobuf/v1/snapshot.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+
+import "common.proto";
+package mayastor.v1;
+
+/// Represenation of a single snapshot. A list this object is to
+/// be returned as part of list snapshots request.
+message Snapshot {
+    string snapshot_uuid = 1; // uuid of the snapshot.
+    string snapshot_name = 2; // name of the snapshot.
+    uint64 snapshot_size = 3; // amount of bytes referenced by snapshot.
+    uint64 num_clones = 4; // number of clones created from this snapshot.
+    google.protobuf.Timestamp timestamp = 5; // snapshot creation_timestamp.
+    string replica_uuid = 6; // uuid of the replica this snapshot is taken from.
+    uint64 replica_size = 7; // amount of bytes referenced by replica.
+    string entity_id = 8; // uuid of the entity for which snapshot is taken.
+    string txn_id = 9; // unique transaction id for snapshot.
+    bool valid_snapshot = 10; // true: valid, false: invalid (missing one/more snapshotdescriptor
+                              //field i.e. txn_id, entity_id, replica_uuid.
+}
+
+/// List Snapshots Request for an object.
+message ListSnapshotsRequest {
+    optional string object_uuid = 1;  // uuid of the object to list snapshots for.
```

Comment:
@hrudaya21 @mtzaurus Why is this input uuid kept optional originally?

Reply:
> @hrudaya21 @mtzaurus Why is this input uuid kept optional originally?

Before I was unware of the `optional` field and handling it based on field length. Now it is handled based on `optional` property.

Reply:
Sorry, I meant to ask why is this input optional? Shouldn't this be mandatory to provide either nexus uuid or replica uuid to list snapshots for.

Reply:
The idea for this parameter was, if the uuid is not passed, then all the snapshot present in the system will be respond back. If uuid is passed, then uuid based filteration will happen for the snapshot and only the snapshots whose parent is same as input uuid from CP will be returned back.

Reply:
hmm, I hope `all the snapshot present in the system` means all snapshots for a particular volume from all its replicas? Spanning beyond a volume(if possible at all) will be an overkill.